### ISSUE TITLE
0.13: [FEATURE/Bug]: Native support for running arbitrary commands on file change or improved compatibility with tools like `entr` 

### ISSUE THREAD

**Issue by worldofgeese** [garden-io/garden#4058]
### Current Behavior

When using 0.12's Dev Mode or Bonsai's Code Synchronization feature with [entr](https://github.com/eradman/entr), `entr` reloads when Mutagen has synced one file. If multiple files are specified, `entr` will only reload on first file sync.

### Expected behavior

`entr` should reload with subsequent file syncs.

If this issue is treated as a feature request, native support for arbitrary actions on file change could be considered.

This is a bug if solving the problem of `entr` only reloading on first file sync is treated. Prior art exists in our [quickstart example](https://github.com/garden-io/quickstart-example/blob/main/api/garden.yml#L12) for single file actions.

This issue was [originally reported](https://discord.com/channels/817392104711651328/1085882625312432179/1088449255880019988
) by @pepov of Axoflow. @worldofgeese will validate with a reproducible example.

### Reproducible example

WIP

### Your environment

* **OS:** WIP
* **How I'm running Kubernetes:** minikube

`garden version` 0.13.0

**Comment by edvald** [comment]
I don't believe this is a bug in Garden (seems to be more to do with how entr responds to fiel changes), but it is certainly a valid feature request—that we've actually explored—to allow arbitrary command execution post sync.

**Comment by pepov** [comment]
I don't consider this as a bug in Garden either. I'm not sure what would be a good solution here, but I think a manual one-time `sync` command would be nice in the new interactive mode.
